src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -574,15 +574,10 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> _PATH:
             return self.dirpath
 
         if len(trainer.loggers) > 0:
-            if trainer.loggers[0].save_dir is not None:
```

Comment:
I think we should keep this and remove `save_dir = trainer.loggers[0].save_dir or trainer.default_root_dir` which makes save_dir with empty string "" work fines.

Reply:
True. @tshu-w Can you send a patch?

Reply:
Sure, https://github.com/Lightning-AI/lightning/pull/15638